tests/queries/0_stateless/02725_async_insert_table_setting.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} -n --query "
+DROP TABLE IF EXISTS t_mt_async_insert;
+DROP TABLE IF EXISTS t_mt_sync_insert;
+
+CREATE TABLE t_mt_async_insert (id UInt64, s String)
+ENGINE = MergeTree ORDER BY id SETTINGS async_insert = 1;
+
+CREATE TABLE t_mt_sync_insert (id UInt64, s String)
+ENGINE = MergeTree ORDER BY id SETTINGS async_insert = 0;"
+
+url="${CLICKHOUSE_URL}&async_insert=0&wait_for_async_insert=1"
```

Comment:
Why is it `.sh` test? Can't we set `wait_for_async_insert` for queries via native protocol?

Reply:
See the note in the changelog. The table-level setting doesn't work in `clickhouse-client` because client executes INSERT queries differently depending on `async_insert` setting. For sync inserts it parses the data by itself and sends it via native protocol and async insert in inapplicable in this case. Getting the value of table-level setting in hard on client.